File: samples/perl/04primality/prime.pl
#!/usr/bin/perl
use warnings;
use strict;

use Math::Complex;

sub IsPrime {
    my ($number) = @_;
    my $limit = sqrt($number);
    for (2 .. $limit) {
        if ($number % $_ == 0) {
            return 0;
        }
    }

    return 1;
}


my $len = scalar(@ARGV);
if ($len == 0) {
    print STDERR "Error: Expected to receive $0 <number> [<number> ...]\n";
    exit(1);
}


foreach my $temp (@ARGV) {
    my $number = int($temp) or die "Expected to receive a number. Received '$temp'\n";

    my $prime = IsPrime($number) ? "true" : "false";
    print("$number => $prime\n");
}

File: samples/typescript/04primality/prime.ts
function isPrime(number: number) {
    const limit: number = Math.sqrt(number);
    for (let i = 2; i <= limit; i++) {
        if (number % i === 0) {
            return false;
        }
    }
    return true;
}

if (require.main === module) {
    if (process.argv.length < 3) {
        console.error('Error: Usage node prime.js <number> [<number> ...]');
        process.exit(1);
    }
    process.argv.slice(2)
        .map(string => Number(string))
        .map(n => `${n} => ${isPrime(n)}`)
        .forEach(str => console.log(str));
}